File: src/main/java/net/hectus/neobb/turn/TCauldron.java
package net.hectus.neobb.turn;

import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.attributes.clazz.NeutralClazz;
import net.hectus.neobb.turn.attributes.function.AttackFunction;
import net.hectus.neobb.turn.attributes.usage.BlockUsage;
import org.bukkit.Material;
import org.bukkit.block.Block;
import org.bukkit.inventory.ItemStack;

public class TCauldron extends Turn<Block> implements BlockUsage, AttackFunction, NeutralClazz {
    public TCauldron(NeoPlayer player) { super(null, player); }
    public TCauldron(Block data, NeoPlayer player) { super(data, player); }

    @Override
    public ItemStack item() {
        return new ItemStack(Material.CAULDRON);
    }

    @Override
    public int cost() {
        return 5; // TODO: Set actual cauldron cost.
    }

    @Override
    public Block getValue() {
        return data;
    }
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/default_game/block/TCauldron.kt
package net.hectus.neobb.modes.turn.default_game.block

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.modes.turn.default_game.attribute.clazz.NeutralClazz
import net.hectus.neobb.modes.turn.default_game.attribute.function.AttackFunction
import net.hectus.neobb.player.NeoPlayer
import org.bukkit.block.Block

class TCauldron(data: Block?, cord: Cord?, player: NeoPlayer?) : BlockTurn(data, cord, player), AttackFunction, NeutralClazz {
    override val cost: Int = 2
}
